tests/unit/s2n_session_ticket_test.c

```diff
@@ -336,6 +336,62 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_config_free(client_config));
     };
 
+    /* Client sends non-empty ST extension. Server does an abbreviated handshake without issuing NST. */
```

Comment:
Can you give this test a better name? This comment should describe the scenario you're trying to test. So something like "Server can decrypt session ticket with key before the key's introduction time, but will not issue new ticket".

I'm also wondering if this test is overkill. Maybe we don't need a full self-talk test. If you don't care about the actual handshake (like how the client and server exchange tickets / issue new tickets), and you're only interested in whether the ticket can be decrypted, you could probably just test s2n_decrypt_session_ticket directly (in s2n_resume_test.c)

Reply:
Will do on the naming.

And I can move the test to `s2n_decrypt_session_ticket`, but I don't know that it ends up being simpler. 

For those tests the server is issuing the tickets, so I either need two separate configs with different key configurations (and therefore need two different connections) or I need to do some time-reversal black-magic clock-mockery to test my "decrypt before encryption time". Out of those options, I think I prefer the multiple config approach. Thoughts?